apps/main/src/lend/entities/market-onchain-rate.ts

```diff
@@ -0,0 +1,32 @@
+import { lendingJsValidationSuite } from '@/lend/entities/validation/lending-js-validation'
+import { ChainId } from '@/lend/types/lend.types'
+import type { ChainParams, ChainQuery } from '@ui-kit/lib/model/query'
+import { queryFactory } from '@ui-kit/lib/model/query'
+import { useApiStore } from '@ui-kit/shared/useApiStore'
+
+type Params = ChainParams<ChainId> & {
+  marketId: string
+}
```

Comment:
There should be a distinction between the query (after validation) and params (before validation)
```suggestion
type MarketQuery = ChainQuery<ChainId> & { marketId: string }
type MarketParams = FieldsOf<MarketQuery>
```